node/src/actors/session/handlers.rs

```diff
@@ -681,19 +681,32 @@ fn inventory_process_superblock(
 
 /// Function to process an InventoryAnnouncement message
 fn inventory_process_inv(session: &mut Session, inv: &InventoryAnnouncement) {
-    // Check how many of the received inventory vectors need to be requested
-    let inv_entries = &inv.inventory;
-
     if !session.requested_block_hashes.is_empty() {
         log::warn!("Received InventoryAnnouncement message while processing an older InventoryAnnouncement. Will stop processing the old one.");
     }
 
-    session.requested_block_hashes = inv_entries
+    let mut limit = usize::try_from(session.config.connections.requested_blocks_batch_limit)
+        .unwrap_or(usize::MAX);
+    if limit == 0 {
+        // unlimited
+        limit = usize::MAX;
+    }
+
+    // Small limits break the synchronization, so force a minimum value of 2 times the superblock
+    // period
+    let min_limit = 2 * usize::from(session.config.consensus_constants.superblock_period);
```

Comment:
After testing a bit I decided to set the minimum limit to 2 times the superblock period. Now any values smaller than that are silently overwritten.

Reply:
The issue with smaller batch size is that the `AddBlocks` handler does not request additional blocks after reaching the target superblock.